### ISSUE TITLE
c++20 warnings and amd gpu 

### ISSUE THREAD

**Issue by mmarchywka** [shankar1729/jdftx#378]
I bought a little cybergeek mini with 32Gb and an AMD GPU.
Is there any simple way for jdftx to use this GPU?
Also I got a few warning  fwiw , 
 g++ --version
g++ (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0

`[ 85%] Linking CXX shared library libjdftx.so
lto-wrapper: warning: using serial compilation of 80 LTRANS jobs
lto-wrapper: note: see the ‘-flto’ option documentation for more information
[ 85%] Built target jdftxlibCMakeFiles/wannier.dir/wannier/WannierMinimizer_save.cpp.o
...

[ 94%] Building CXX object CMakeFiles/wannier.dir/wannier/commands.cpp.o
In file included from /home/ubuntu/dev/jdftx/jdftx-1.7.0/jdftx/commands/minimize.h:30,
                 from /home/ubuntu/dev/jdftx/jdftx-1.7.0/jdftx/wannier/commands.cpp:20:
/home/ubuntu/dev/jdftx/jdftx-1.7.0/jdftx/commands/command.h:73:26: warning: identifier ‘requires’ is a keyword in C++20 [-Wc++20-compat]
   73 |         std::set<string> requires; //!< Names of other commands that this one requires; those commands will be processed before this one.
      |                          ^~~~~~~~
[ 95%] Building CXX object CMakeFiles/wannier.dir/wannier/main.cpp.o
[ 96%] Linking CXX executable wannier
lto-wrapper: warning: using serial compilation of 8 LTRANS jobs
lto-wrapper: note: see the ‘-flto’ option documentation for more information
[ 96%] Built target wannier
[ 97%] Building CXX object CMakeFiles/phonon.dir/phonon/Phonon.cpp.o
[ 97%] Building CXX object CMakeFiles/phonon.dir/phonon/Phonon_init.cpp.o
[ 98%] Building CXX object CMakeFiles/phonon.dir/phonon/Phonon_supercell.cpp.o
In file included from /home/ubuntu/dev/jdftx/jdftx-1.7.0/jdftx/phonon/Phonon_supercell.cpp:22:
/home/ubuntu/dev/jdftx/jdftx-1.7.0/jdftx/commands/command.h:73:26: warning: identifier ‘requires’ is a keyword in C++20 [-Wc++20-compat]
   73 |         std::set<string> requires; //!< Names of other commands that this one requires; those commands will be processed before this one.
      |                          ^~~~~~~~
[ 98%] Building CXX object CMakeFiles/phonon.dir/phonon/commands.cpp.o
`

**Comment by shankar1729** [comment]
No, there is no easy way to use AMD GPUs at the moment, and as discussed in the other thread, it is not necessarily a simple matter to get this working with hipify.
